src/comsock.c

```diff
@@ -543,3 +543,39 @@ natsSock_InitDeadline(natsSockCtx *ctx, int64_t timeout)
     natsDeadline_Init(&ctx->readDeadline, timeout);
     natsDeadline_Init(&ctx->writeDeadline, timeout);
 }
+
+void
+natsSock_GetLocalIPAndPort(natsSockCtx *ctx, char **ip, int *port)
+{
+    struct sockaddr_storage addr;
+    natsSockLen             addrLen = (natsSockLen) sizeof(addr);
+    char                    localIP[64];
+    void                    *laddr = NULL;
+    int                     fam;
+
+    *ip = NULL;
+    *port = 0;
+
+    if (getsockname(ctx->fd, (struct sockaddr*) &addr, &addrLen) != 0)
+        return;
```

Comment:
Should this method return a NATS_OK by default, or an error here?

Reply:
At first I was doing that inside the connect and did not want to fail a connection just because of error getting the socket name, etc.. but since I have made that a separate call now, I will, thanks.

Reply:
I see, yeah this seems like a better place than the connect